sdks/python/src/opik/evaluation/metrics/llm_judges/usefulness/metric.py

```diff
@@ -0,0 +1,128 @@
+        >>> result = usefulness_metric.score("What's the capital of France?", "The capital of France is Paris.")
+        >>> print(result.value)  # A float between 0.0 and 1.0
+        >>> print(result.reason)  # Explanation for the score
+    """
+
+    def __init__(
+        self,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "UsefulnessMetric",
+        track: bool = True,
+    ):
+        super().__init__(
+            name=name,
+            track=track,
+        )
+
+        self._init_model(model)
+
+    def _init_model(
+        self, model: Optional[Union[str, base_model.OpikBaseModel]]
+    ) -> None:
+        if isinstance(model, base_model.OpikBaseModel):
+            self._model = model
+        else:
+            self._model = models_factory.get(model_name=model)
+
+    def score(
+        self, input: str, output: str, **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Calculate the usefulness score for the given input-output pair.
+
+        Args:
+            input: The input text to be evaluated.
+            output: The output text to be evaluated.
+            **ignored_kwargs: Additional keyword arguments that are ignored.
+
+        Returns:
+            score_result.ScoreResult: A ScoreResult object containing the usefulness score
+            (between 0.0 and 1.0) and a reason for the score.
+        """
+        llm_query = template.generate_query(
+            input=input,
+            output=output,
+        )
+        model_output = self._model.generate_string(
+            input=llm_query, response_format=UsefulnessResponseFormat
+        )
+
+        return self._parse_model_output(model_output)
+
+    async def ascore(
+        self, input: str, output: str, **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Asynchronously calculate the usefulness score for the given input-output pair.
+
+        Args:
+            input: The input text to be evaluated.
+            output: The output text to be evaluated.
+            **ignored_kwargs: Additional keyword arguments that are ignored.
+
+        Returns:
+            score_result.ScoreResult: A ScoreResult object containing the usefulness score
+            (between 0.0 and 1.0) and a reason for the score.
+        """
+        llm_query = template.generate_query(
+            input=input,
+            output=output,
+        )
+        model_output = await self._model.agenerate_string(
+            input=llm_query, response_format=UsefulnessResponseFormat
+        )
+
+        return self._parse_model_output(model_output)
+
+    def _parse_model_output(self, content: str) -> score_result.ScoreResult:
+        """Parse the model output string into a ScoreResult."""
+        try:
+            dict_content = json.loads(content)
+            score: float = float(dict_content["score"])
+
+            if not (0.0 <= score <= 1.0):
+                score = 0.5
```

Comment:
I know that historically we have other metrics with a similar rule, but I don't want us to continue doing that (we'll need to update the rest of them soon I suppose).
I think it'll be better to raise `MetricComputationError` in that case, we essentially failed to parse the score and I don't think it's a good idea to guess that the result was somewhere between good and fair.

Reply:
Good point, I removed it